WordPress/Classes/Services/PostCoordinator.swift

```diff
@@ -109,6 +111,50 @@ class PostCoordinator: NSObject {
         save(post)
     }
 
+    // TODO: Add notices
+    // TODO: Remove isFirstTimePublish need
+    // TODO: Check correct error is displayed fofr network error
+
+    /// Publishes the post according to the current settings and user capabilities.
+    ///
+    /// - warning: Work-in-progress (kahu-offline-mode)
+    @MainActor
+    func _publish(_ post: AbstractPost) async throws {
+        let parameters = PostHelper.remotePost(with: post)
+        if post.status == .draft {
+            parameters.status = PostStatusPublish
+            parameters.date = Date()
+        } else {
+            // Pulbish according to the currrent post settings: private, scheduled, etc.
+        }
+        do {
+            let repository = PostRepository(coreDataStack: coreDataStack)
+            try await repository._upload(parameters, for: post)
+            didPublish(post)
+
+            let notice = PostNoticeViewModel.makeSuccessNotice(for: post, isFirstTimePublish: true)
+            actionDispatcherFacade.dispatch(NoticeAction.post(notice))
+        } catch {
+            let notice = PostNoticeViewModel.makeFailureNotice(for: post, error: error)
+            actionDispatcherFacade.dispatch(NoticeAction.post(notice))
+
+            throw error
+        }
+    }
+
+    private func didPublish(_ post: AbstractPost) {
+        if post.status == .publish {
+            QuickStartTourGuide.shared.complete(tour: QuickStartPublishTour(), silentlyForBlog: post.blog)
+        }
+        if post.isScheduled() {
+            notifyNewPostScheduled()
+        } else if post.isPublished() {
+            notifyNewPostPublished()
+        }
+        SearchManager.shared.indexItem(post)
+        AppRatingUtility.shared.incrementSignificantEvent()
```

Comment:
Everything non-editor related is now handled by `PostCoordinator` to allow you to safely publish outside of the editor.

Noting that it no longer sets `post.shouldAttemptAutoUpload = true` and no longer uses `isFirstTimePublish` which should never have been part of the model layer.